Harmony Profile Image adjustment

JackWillDavis opened the pull request:
This PR contains a tweak to the harmony discussion core which 

Removing the profile image from discussion posts and replacing it with the username of the poster. Default link added to make it match in with local styling.

It definitely a bit hacky, adding a new user_load now I look at it again but if there is a better way of grabbing the username of the poster then please let me know :)

JackWillDavis commented:
As discussed in issue #2357 with @btopro and @rickhumphries

btopro commented:
@JackWillDavis can you also apply it to the file in question and commit that? I appreciate you have the patch here but we're good with accepting this change as well so you don't have to run patched (patch helps us track in the event we ever upgrade that aspect)

JackWillDavis commented:
OK, I'll just make the change now

JackWillDavis commented:
All done!

btopro commented:
so this will work in your setup but in 0.10.x we should use the function that helps resolve names as it's cleaner and allows for integration w/ remote services / alternate name providing services like SIS -- `_elmsln_core_get_user_name()`

btopro commented:
https://github.com/elmsln/elmsln/blob/0.10.x/core/dslmcode/shared/drupal-7.x/modules/elmsln_contrib/cis_connector/modules/features/elmsln_core/elmsln_core.module#L2300 -- this will figure out the name to provide with a fallback of ->name

JackWillDavis commented:
OK, updated and neatened up a bit. Let me know if this is OK. I've updated the patch file but not sure if you wanted changes to files AND a patch file but it's all there now.

btopro commented:
nice! is there a reason for wrapping the name in a link tag?

JackWillDavis commented:
It was so it didn't need to be restyled in CSS. If it's wrapped in a link, it keeps the same design for the username as used elsewhere in the post template.

btopro commented:
ahhh ok. can you through a `tabindex="-1"` on that to improve accessibility then since it's just purely for styling

JackWillDavis commented:
I'll remember that for the future :) Added now.

btopro commented:
for future A11Y reference - that tabindex tells the screenreader / you tabbing through on the keyboard, to ignore that link. This is useful when things that would otherwise get focus shouldn't get them for whatever reason

JackWillDavis commented:
I think I remember using tabindex years ago for ignoring links on an advertising slider but I'll bear accessibility in mind for PRs here.
